runtime/lua/vim/shared.lua

```diff
@@ -544,14 +544,9 @@ function vim.spairs(t)
   end
   table.sort(keys)
 
-  -- Return the iterator function.
-  -- TODO(justinmk): Return "iterator function, table {t}, and nil", like pairs()?
-  local i = 0
+  local it = vim.iter(t)
   return function()
-    i = i + 1
-    if keys[i] then
-      return keys[i], t[keys[i]]
-    end
+    return it:next()
```

Comment:
I realize this is closed, but just want to mention that it's not necessary to return

```lua
function()
  return it:next()
end
```

You can simply return `it`, since the `Iter` metatable has a `__call` metamethod (which calls `:next()`).